typescript/src/v0/Mcap0UnindexedWriter.ts

```diff
@@ -0,0 +1,60 @@
+import { IWritable } from "./IWritable";
+import { Mcap0RecordWriter } from "./Mcap0RecordWriter";
+import { ChannelInfo, Message, Header, Attachment } from "./types";
+
+export class Mcap0UnindexedWriter {
+  private recordWriter: Mcap0RecordWriter;
+
+  private channelInfos = new Map<number, ChannelInfo>();
+  private writtenChannelIds = new Set<number>();
+
+  constructor(writable: IWritable) {
+    this.recordWriter = new Mcap0RecordWriter(writable);
+  }
+
+  async start(header: Header): Promise<void> {
+    await this.recordWriter.writeMagic();
+    await this.recordWriter.writeHeader(header);
+  }
+
+  async end(): Promise<void> {
+    await this.recordWriter.writeFooter({
+      indexOffset: 0n,
+      indexCrc: 0,
+    });
+  }
+
+  /**
+   * Add channel info and return a generated channel id. The channel id is used when adding messages.
+   */
+  async registerChannel(info: Omit<ChannelInfo, "channelId">): Promise<number> {
```

Comment:
thoughts on `addChannel` vs `registerChannel`? I used `addChannel` for ws-protocol stuff.

Reply:
I had addChannel and switched to registerChannel since it returns an ID - don't feel strongly tho